### PR TITLE
cmd profile support get by name or address list

### PR DESCRIPTION
**Pull request by tongjingran**
Related to [#66](https://github.com/qiniu/goc/issues/66)

### CODE REVIEW

**Comment by CarlJi** [comment]
This PR fixes #41 also.

**Comment by CarlJi** [comment]
Please also update Tips with such points to give user clear message that they can get profiles as choice.

File: pkg/cover/server.go [hunk 1]
```diff
@@ -67,7 +69,7 @@ func GocServer(w io.Writer) *gin.Engine {
 	v1 := r.Group("/v1")
 	{
 		v1.POST("/cover/register", registerService)
-		v1.GET("/cover/profile", profile)
```
Comment (lyyyuna) [inline pkg/cover/server.go:70]: why POST, not compatible with RESTful API

**Comment by lyyyuna** [comment]
> This PR fixes #41 also.

@CarlJi 
for #41 
```
			pp, err := NewWorker(addr).Profile()
			if err != nil {
				c.JSON(http.StatusExpectationFailed, gin.H{"error": err.Error()})
				return
			}
```

it will still return error, if one of the service unreachable

File: pkg/cover/server.go (hunk 1 above)
Comment (tongjingran) [reply pkg/cover/server.go:70]: "name" and "address" are long, and may have '&' in it. Use post method more suitable in this situation.

File: pkg/cover/server.go [hunk 2]
```diff
@@ -120,11 +129,61 @@ func registerService(c *gin.Context) {
...
 func profile(c *gin.Context) {
-	svrsUnderTest := DefaultStore.GetAll()
+	respByte, err := ioutil.ReadAll(c.Request.Body)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, err.Error())
+		return
+	}
+	var param ProfileParam
+	json.Unmarshal(respByte, &param)
+	if param.Name != "" && param.Address != "" {
```
Comment (lyyyuna) [inline pkg/cover/server.go:139]: param.Name and param.Address cannot be set at the same time?

File: pkg/cover/server.go [hunk 3]
```diff
@@ -120,11 +129,61 @@ func registerService(c *gin.Context) {
...
+	svrsUnderTest := make(map[string][]string)
+	if param.Name == "" && param.Address == "" {
+		svrsUnderTest = svrsAll
+	} else {
+		if param.Name != "" {
+			for _, name := range nameList {
+				miss := true
+				for svr, addrs := range svrsAll {
+					if svr == name {
+						svrsUnderTest[svr] = addrs
```
Comment (lyyyuna) [inline pkg/cover/server.go:155]: How to handle duplicate service name?

Comment (tongjingran) [reply pkg/cover/server.go:154]: if service name is same as before, the map will assign to the struct.

Comment (lyyyuna) [reply pkg/cover/server.go:154]: that's the issue, 
1. if different binary share the same name, how? (the filepath is same, import path is different)
2. if same binary has different services depends on args (like goc), how? (both import path and filepath are same)

**Comment by tongjingran** [comment]
The PR fixes https://github.com/qiniu/goc/issues/72 also

File: cmd/profile.go [hunk 4]
```diff
@@ -35,17 +35,25 @@ var profileCmd = &cobra.Command{
...
-goc profile -o ./coverage.cov
-
-# Get coverage counter from specified register center, the result output to specified file.
-goc profile --center=http://192.168.1.1:8080 -o ./coverage.cov
-
 # Get coverage counter from specified register center, the result output to specified file.
 goc profile --center=http://192.168.1.1:8080 --output=./coverage.cov
+
+# Get coverage counter of several specified services
+goc profile --service=service1,service2 --service=service3
```
Comment (CarlJi) [inline cmd/profile.go:42]: suggest change to 
`goc profile --service=service1,service2,service3`

File: pkg/cover/server.go [hunk 5]
```diff
@@ -194,3 +225,58 @@ func convertProfile(p []byte) ([]*cover.Profile, error) {
...
+		}
+	}
+	return false
+}
+
+// getSvrUnderTest get service map by service and address list
+func getSvrUnderTest(svrList, addrList []string, force bool, svrsAll map[string][]string) (svrsUnderTest map[string][]string, err error) {
+	svrsUnderTest = map[string][]string{}
+	if len(svrList) != 0 && len(addrList) != 0 {
+		return nil, fmt.Errorf("use this flag and 'address' flag at the same time is illegal")
```
Comment (CarlJi) [inline pkg/cover/server.go:242]: unclear message

**Pull request merged**